### DISCUSSION TITLE
phoebe-server problem

### DISCUSSION THREAD

**Discussion by esborowski** [phoebe-project/phoebe2#646] [Questions]
Hello, all,

I hope this is the right place for this discussion. I am new to phoebe and as such would love to get the gui working so I can familiarize myself a bit more easily. I would happily use the web client, but it seems the server is down. So, I have the desktop client installed, but when launched, it can't start the local server. To check that phoebe-server was working, I called it in the terminal, but ended up with this error:
```
eric@eblinlap:~$ phoebe-server
Traceback (most recent call last):
  File "/home/eric/.local/bin/phoebe-server", line 19, in <module>
    from flask_socketio import SocketIO, emit, join_room, leave_room
  File "/home/eric/.local/lib/python3.10/site-packages/flask_socketio/__init__.py", line 24, in <module>
    from werkzeug.serving import run_with_reloader
ImportError: cannot import name 'run_with_reloader' from 'werkzeug.serving' (/home/eric/.local/lib/python3.10/site-packages/werkzeug/serving.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/eric/.local/bin/phoebe-server", line 22, in <module>
    raise ImportError("dependencies not met: pip install flask flask-cors flask-socketio gevent-websocket")
ImportError: dependencies not met: pip install flask flask-cors flask-socketio gevent-websocket

```
My first try at a solution was to pip install flask flask-cors flask-socketio gevent-websocket, but it seems I have these and they're all up to date. I am not sure what to try next. 
Thanks!

**Comment by kecnry** [comment]
Yes, sorry, the public server has been down for sometime and unfortunately can't provide an ETA for when we'll get the problem fixed. 
 It looks like the original error you're seeing is coming directly from `flask_socketio` (you can ignore the error about dependencies not met - that is just being raised because `flask_socketio` failed to import, but in your case it failed to import because of some issue with itself and not because you don't have it installed).

Can you try running that line on its own (`python -c "from flask_socketio import SocketIO, emit, join_room, leave_room"`) and see if it gives you the same error?  Can you also let us know the version of `flask_socketio` you have installed?

-Kyle

**Comment by esborowski** [comment]
Thank you for the swift reply! 
Here is the result of running that line:
```
eric@eblinlap:~$ python3 -c "from flask_socketio import SocketIO, emit, join_room, leave_room"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/eric/.local/lib/python3.10/site-packages/flask_socketio/__init__.py", line 24, in <module>
    from werkzeug.serving import run_with_reloader
ImportError: cannot import name 'run_with_reloader' from 'werkzeug.serving' (/home/eric/.local/lib/python3.10/site-packages/werkzeug/serving.py)
```

Running pip install flask_socketio returns 
`Requirement already satisfied: flask_socketio in ./.local/lib/python3.10/site-packages (4.3.2)
`
among other things. Does this indicate the version is 4.3.2?

**Reply by kecnry** [reply]
Perfect, thanks (yes it looks like you have version 4.3.2 which should be compatible with phoebe).  

I found [other reports online of the same issue](https://stackoverflow.com/questions/71850969/importerror-cannot-import-name-run-with-reloader-from-werkzeug-serving) and it looks like the general consensus is to either upgrade `flask_socketio` (although phoebe has only been tested with 4.3.x, so that might not work), or possibly to install an older version of werkzeug (apparently below 2.1).  My suggestion would be to try manually downgrading your installation of werkzeug first as long as you don't have anything else that requires the newer version.  Let me know if that works and I can put the version requirement into the phoebe installation so other people won't run into this.

**Reply by esborowski** [reply]
I downgraded werkzeug to 2.0, but then received an error that my version of flask doesn't work with werkzeug 2.0. I got around this by downgrading flask to 2.0 first, then downgrading werkzeug to 2.0. Now, when I launch the desktop ui, it starts a local server as it should.

Thank you for your help. Clear skies!

**Reply by kecnry** [reply]
That's great to hear!  Can you let me know the version of flask you had to downgrade to (and the version that didn't work, if you remember) and then I'll try to update the requirements (I opened #647 to track that effort).  Thanks!

**Reply by esborowski** [reply]
Sorry for the delayed reply. I don't recall the version number that didn't work, but my flask was up to date when I was having this issue. I downgraded to version 2.0.0 for both flask and werkzeug.
